sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Processor/EventProcessorClientLiveTests.cs

```diff
@@ -382,75 +278,71 @@ await using (var connection = new EventHubConnection(connectionString))
         /// </summary>
         ///
         [Test]
-        [Ignore("Unstable test. (Tracked by: #7458)")]
-        public async Task LoadBalancingIsEnforcedWhenDistributionIsUneven()
+        [TestCase(2)]
+        [TestCase(4)]
+        [TestCase(15)]
+        public async Task EventProcessorWaitsMaximumWaitTimeForEvents(int maximumWaitTimeInSecs)
         {
-            var partitions = 10;
-
-            await using (EventHubScope scope = await EventHubScope.CreateAsync(partitions))
-            {
-                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
-
-                await using (var connection = new EventHubConnection(connectionString))
-                {
-                    ConcurrentDictionary<string, int> ownedPartitionsCount = new ConcurrentDictionary<string, int>();
-
-                    // Create the event processor manager to manage our event processors.
-
-                    var eventProcessorManager = new EventProcessorManager
-                        (
-                            EventHubConsumerClient.DefaultConsumerGroupName,
-                            connectionString
-                        // TODO: fix test. OwnerIdentifier is not accessible anymore.
-                        // onInitialize: eventArgs =>
-                        //     ownedPartitionsCount.AddOrUpdate(eventArgs.Context.OwnerIdentifier, 1, (ownerId, value) => value + 1),
-                        // onStop: eventArgs =>
-                        //     ownedPartitionsCount.AddOrUpdate(eventArgs.Context.OwnerIdentifier, 0, (ownerId, value) => value - 1)
-                        );
-
-                    eventProcessorManager.AddEventProcessors(1);
-
-                    // Start the event processors.
-
-                    await eventProcessorManager.StartAllAsync();
-
-                    // Make sure the event processors have enough time to stabilize.
-
-                    await eventProcessorManager.WaitStabilization();
-
-                    // Assert all partitions have been claimed.
-
-                    Assert.That(ownedPartitionsCount.ToArray().Single().Value, Is.EqualTo(partitions));
+            await using EventHubScope scope = await EventHubScope.CreateAsync(2);
+            var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+            await using var connection = new EventHubConnection(connectionString);
+            var timestamps = new ConcurrentDictionary<string, ConcurrentBag<DateTimeOffset>>();
+            using var cancellationSource = new CancellationTokenSource();
+            cancellationSource.CancelAfter(TimeSpan.FromSeconds(30));
+            var receivedCount = 0;
+
+            // Create the event processor manager to manage our event processors.
+
+            var eventProcessorManager = new EventProcessorManager
+                (
```

Comment:
nit:  We normally don't indent the parenthesis, but I don't have a concern if you decide to leave these as-is.